File: src/main/java/com/noximity/remmyChat/utils/PlaceholderUtils.java
package com.noximity.remmyChat.utils;

import me.clip.placeholderapi.PlaceholderAPI;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class PlaceholderUtils {

    private static final boolean placeholderApiEnabled = Bukkit.getPluginManager().getPlugin("PlaceholderAPI") != null;

    private PlaceholderUtils() {
    }

    public static String applyPlaceholders(Player player, String text) {
        if (placeholderApiEnabled && player != null) {
            return PlaceholderAPI.setPlaceholders(player, text);
        }
        return text;
    }

    public static boolean isPlaceholderApiEnabled() {
        return placeholderApiEnabled;
    }
}

File: bin/main/com/noximity/remmyChat/utils/PlaceholderUtils.kt
package com.noximity.remmyChat.utils

import me.clip.placeholderapi.PlaceholderAPI
import org.bukkit.Bukkit
import org.bukkit.entity.Player

object PlaceholderUtils {
    val isPlaceholderApiEnabled: Boolean = Bukkit.getPluginManager().getPlugin("PlaceholderAPI") != null

    fun applyPlaceholders(player: Player?, text: String): String? {
        if (isPlaceholderApiEnabled && player != null) {
            return PlaceholderAPI.setPlaceholders(player, text)
        }
        return text
    }
}